spring-kafka/src/main/java/org/springframework/kafka/retrytopic/RetryTopicConfigurationSupport.java

```diff
@@ -64,13 +63,20 @@
  * {@link EnableKafkaRetryTopic @EnableRetryTopic}.
  *
  * @author Tomaz Fernandes
+ * @author Gary Russell
  * @since 2.9
 */
 public class RetryTopicConfigurationSupport {
 
+	private static final String BACK_OFF_MANAGER_THREAD_EXECUTOR_BEAN_NAME = "backOffManagerThreadExecutor";
+
+	private static final AtomicBoolean ONLY_ONE_ALLOWED = new AtomicBoolean(true);
+
 	private final RetryTopicComponentFactory componentFactory = createComponentFactory();
 
-	private static final String BACK_OFF_MANAGER_THREAD_EXECUTOR_BEAN_NAME = "backOffManagerThreadExecutor";
+	protected RetryTopicConfigurationSupport() {
+		Assert.state(ONLY_ONE_ALLOWED.getAndSet(false), "Only one 'RetryTopicConfigurationSupport' is allowed");
```

Comment:
I wonder if there still exist multi-application solutions in a single JVM out there.
Like Tomcat, JBOSS AS etc..
Kinda this `static` flag would be a show stopper, if several spring-kafka application are ran in the same JVM. 😸 .

Not sure though that I have some other solution. Perhaps check for some bean in the application context...

Reply:
I believe they would be in different class loaders there. But we can re-think this after the RC.